src/compressor/zstd/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# zstd
+
+# libzstd - build it statically
+set(ZSTD_EXTRA_CFLAGS -fPIC -Wno-unused-variable -O3)
+
+add_custom_target(build_zstd
+  COMMAND $(MAKE) libzstd CFLAGS="${ZSTD_EXTRA_CFLAGS}"
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/zstd/lib
+  COMMENT "libzstd building")
+
+add_library(zstd STATIC IMPORTED)
+set_property(TARGET zstd PROPERTY IMPORTED_LOCATION "${CMAKE_SOURCE_DIR}/src/zstd/lib/libzstd.a")
```

Comment:
this put the built asserts into the source directory. maybe we can put them into the binary directory? see https://github.com/tchaikov/ceph/commit/071562e6e961016bbc9254ef57d1d20e6c2b4f70
the patch follows the example of how we build rocksdb, so we can be more consistent when including external projects.